src/agent/TransportService.ts

```diff
@@ -0,0 +1,94 @@
+import { Logger } from '../logger'
+import { ConnectionRecord, ConnectionRole } from '../modules/connections'
+
+const DID_COMM_TRANSPORT_QUEUE = 'didcomm:transport/queue'
+
+export class TransportService {
+  private transportTable: TransportTable = {}
+  private logger: Logger
+
+  public constructor(logger: Logger) {
+    this.logger = logger
+  }
+
+  public saveTransport(connectionId: string, transport: Transport) {
+    this.transportTable[connectionId] = transport
+  }
+
+  public resolveTransport(connection: ConnectionRecord): Transport {
+    const transport = this.getTransport(connection.id)
+    if (transport) {
+      return transport
+    }
+
+    const endpoint = this.findEndpoint(connection)
+    if (endpoint) {
+      if (endpoint?.startsWith('ws')) {
+        return new WebSocketTransport(endpoint)
+      } else if (endpoint === DID_COMM_TRANSPORT_QUEUE) {
+        return new DidCommQueueTransport()
+      }
+      return new HttpTransport(endpoint)
+    }
+
+    throw new Error(`No transport found for connection with id ${connection.id}`)
+  }
+
+  public getTransport(connectionId: string) {
+    return this.transportTable[connectionId]
+  }
+
+  private findEndpoint(connection: ConnectionRecord) {
+    if (connection.theirDidDoc) {
+      const endpoint = connection.theirDidDoc.service[0].serviceEndpoint
+      if (endpoint) {
+        this.logger.debug('Taking service endpoint from their DidDoc')
+        return endpoint
+      }
+    }
+
+    if (connection.role === ConnectionRole.Invitee && connection.invitation) {
+      const endpoint = connection.invitation.serviceEndpoint
+      if (endpoint) {
+        this.logger.debug('Taking service endpoint from invitation')
+        return endpoint
+      }
+    }
+  }
+}
+
+interface TransportTable {
+  [connectionRecordId: string]: Transport
+}
+
+type TransportType = 'ws' | 'http' | 'queue'
+
+export interface Transport {
+  type: TransportType
+  endpoint: string
+}
+
+export class WebSocketTransport implements Transport {
+  public type: TransportType = 'ws'
```

Comment:
makes for strict type of `ws`:
```suggestion
  public readonly type = "ws"
```

and then (haven't checked syntax, but something like this should be possible):

```
type TransportType = WebSocketTransport['type'] | ....
```

Reply:
Also is `ws` bound to the scheme. In that case, how about `wss`?

Reply:
I mean `ws` just as WebSocket in general.

Reply:
I updated the attribute using `readonly`

Reply:
In that case, maybe `WebSocket` or `WebSocketTransport`? Seems more in line with the other transport type strings we use (e.g. for records)